#!/usr/bin/env python
"""
Add your desired parameters here. All required headers will be generated from this.
!!!IMPORTANT this file must be executable !!!

Use one of these commands to add parameters to your parameter struct.

 def add(self, name, paramtype, description, level=0, edit_method='""', default=None, min=None, max=None, configurable=False, global_scope=False, constant=False):
        Adds parameters to your parameter struct.

        - If no default value is given, you need to specify one in your launch file
        - Global parameters, vectors, maps and constant params can not be configurable
        - Global parameters, vectors and maps can not have a default, min or max value

        :param self:
        :param name: The Name of you new parameter
        :param paramtype: The C++ type of this parameter. Can be any of ['std::string', 'int', 'bool', 'float', 'double'] or std::vector<...> or std::map<std::string, ...>
        :param description: Choose an informative documentation string for this parameter.
        :param level: (optional) Passed to dynamic_reconfigure
        :param edit_method: (optional) Passed to dynamic_reconfigure
        :param default: (optional) default value
        :param min: (optional)
        :param max: (optional)
        :param configurable: (optional) Should this parameter be dynamic configurable
        :param global_scope: (optional) If true, parameter is searched in global ('/') namespace instead of private ('~') ns
        :param constant: (optional) If this is true, the parameter will not be fetched from param server, but the default value is kept.
        :return: None

 def add_enum(self, name, description, entry_strings, default=None):
        Adds an enum to dynamic reconfigure
        :param name: Name of enum parameter
        :param description: Informative documentation string
        :param entry_strings: Enum entries, must be strings! (will be numbered with increasing value)
        :param default: Default value
        :return:
"""
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Self-added
gen.add("SignalMultiplierEnable", bool_t, 0, "")
gen.add("PtpEnable", bool_t, 1, "")
gen.add("PtpSlaveOnly", bool_t, 1, "")
#Default
gen.add("camera_topic", str_t, 0, "name of the camera topic", "/camera/image_raw")

gen.add("GainAuto", str_t, 0, "", "Off")
gen.add("GainSelector", str_t, 0, "", "")
gen.add("ExposureAuto", str_t, 0, "", "Off")
gen.add("TriggerSource", str_t, 0, "", "Software")
gen.add("TriggerActivation", str_t, 0, "", "RisingEdge")
gen.add("TriggerDivider", int_t, 0, "", 1, 1, 64)
gen.add("Line1Source", str_t, 0, "", "Off")

gen.add("PixelFormat", str_t, 0, "", "")
gen.add("DeviceLinkThroughputLimit", int_t, 0, "", 125000000, 1250005, 125000000)

gen.add("selectedDevice", str_t, 0, "", "111111")

gen.add("ExposureTime", int_t, 0, "", 1000, 1000, 99999999)
gen.add("AcquisitionFrameRate", double_t, 0, "", 1.0, 0.0, 250.0)
gen.add("Gamma", double_t, 0, "", 1.0, 0.3, 3.0)

gen.add("ImageWidth", int_t, 0, "", 1, 1, 2500)
gen.add("ImageHeight", int_t, 0, "", 1, 1, 2500)




# DO NOT TOUCH THIS LINE
#Syntax : Package, Node, Config Name(The final name will be TestPylonDetectRosToolParameter)
exit(gen.generate("peak_cam", "peak_cam", "PeakCam"))
